Lib/test/test_types.py

```diff
@@ -951,9 +951,9 @@ def __eq__(self, other):
         with self.assertRaises(ZeroDivisionError):
             list[int] | list[bt]
 
-        union_ga = (int | list[str], int | collections.abc.Callable[..., str],
-                    int | d)
-        # Raise error when isinstance(type, type | genericalias)
+        union_ga = (list[str] | int, collections.abc.Callable[..., str] | int,
+                    d | int)
+        # Raise error when isinstance(type, genericalias | type)
```

Comment:
Just to be clear, this test only broke because genericalias exceptions are now lazy right? So if it sees a valid type first and matches that, no exception will be raised.

Reply:
You are totally right. It will match `tuple` behavior.